### PR TITLE
abide by CCCL config macro naming conventions for `_CCCL_PRETTY_FUNCTION` and `_CCCL_NO_BUILTIN_STRLEN`

### PR DESCRIPTION
**Pull request by ericniebler**
## Description

some CCCL config macros i touched recently (`_CCCL_PRETTY_FUNCTION` and `CCCL_BUILTIN_NO_STRLEN`) don't follow the established naming conventions for such things. this PR:

- renames `_CCCL_PRETTY_FUNCTION` to `_CCCL_BUILTIN_PRETTY_FUNCTION` and makes it a function-like macro, following the lead of #2628.
- replaces `_CCCL_NO_BUILTIN_STRLEN` with `_CCCL_BUILTIN_STRLEN(str)`, which is not defined if there is no such builtin.

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by miscco — APPROVED** [review]

**Review by miscco — APPROVED** [review]

File: libcudacxx/include/cuda/std/detail/libcxx/include/__string [hunk 1]
```diff
@@ -254,19 +254,30 @@ struct _CCCL_TYPE_VISIBILITY_DEFAULT char_traits<char>
 
   _LIBCUDACXX_HIDE_FROM_ABI static _CCCL_CONSTEXPR_CXX14 int
   compare(const char_type* __s1, const char_type* __s2, size_t __n) noexcept;
-  _LIBCUDACXX_HIDE_FROM_ABI static size_t _CCCL_CONSTEXPR_CXX14 length(const char_type* __s) noexcept
+
+  // Define a fallback _CCCL_BUILTIN_STRLEN function using parens to prevent
+  // macro expansion.
+  _LIBCUDACXX_HIDE_FROM_ABI static _CCCL_CONSTEXPR_CXX14 size_t(_CCCL_BUILTIN_STRLEN)(const char_type* __s) noexcept
```
Comment (miscco) [inline libcudacxx/include/cuda/std/detail/libcxx/include/__string:263]: I am strongly against naming this the same. We can name this `__strlen_fallback` but that is a bit much

File: libcudacxx/include/cuda/std/detail/libcxx/include/__string [hunk 2]
```diff
@@ -254,27 +257,31 @@ struct _CCCL_TYPE_VISIBILITY_DEFAULT char_traits<char>
...
+  {
+#if defined(_CCCL_COMPILER_GCC) && _CCCL_GCC_VERSION < 130000
+    // absurd workaround for GCC "internal compiler error: in cxx_eval_array_reference"
+    if (__libcpp_is_constant_evaluated())
+      ;
+#endif
+    // On device, this will always use the fallback function above. On host,
+    // this will use __builtin_strlen if it is available; otherwise, it will
+    // call the fallback function above.
+    NV_IF_ELSE_TARGET(NV_IS_DEVICE, (return (_CCCL_BUILTIN_STRLEN) (__s);), (return _CCCL_BUILTIN_STRLEN(__s);))
```
Comment (miscco) [inline libcudacxx/include/cuda/std/detail/libcxx/include/__string:282]: What speaks against:
```suggestion
#if defined(_CCCL_BUILTIN_STRLEN)
    NV_IF_ELSE_TARGET(NV_IS_DEVICE, (return (_CCCL_BUILTIN_STRLEN) (__s);), (return __strlen_fallback(__s);))
#else // ^^^ _CCCL_BUILTIN_STRLEN ^^^ / vvv !_CCCL_BUILTIN_STRLEN vvv
    return __strlen_fallback(__s);
#endif // !_CCCL_BUILTIN_STRLEN 
```

Comment (ericniebler) [reply libcudacxx/include/cuda/std/detail/libcxx/include/__string:282]: apart from the fact that it is 500% more code, nothing. ;-)

Comment (miscco) [reply libcudacxx/include/cuda/std/detail/libcxx/include/__string]: Thats true, but the 500% are about 5 lines and we dont have to fake a macro invocation

**Pull request merged**